projects/vdk-core/plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import click
+from tabulate import tabulate
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core.config import Configuration
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.plugin.impala.impala_connection import ImpalaConnection
+
+
+def _connection_by_configuration(configuration: Configuration):
+    return ImpalaConnection(
+        host=configuration.get_value("IMPALA_HOST"),
+        port=configuration.get_value("IMPALA_PORT"),
+        database=configuration.get_value("IMPALA_DATABASE"),
+        timeout=configuration.get_value("IMPALA_TIMEOUT"),
+        use_ssl=configuration.get_value("IMPALA_USE_SSL"),
+        ca_cert=configuration.get_value("IMPALA_CA_CERT"),
+        auth_mechanism=configuration.get_value("IMPALA_AUTH_MECHANISM"),
+        user=configuration.get_value("IMPALA_USER"),
+        password=configuration.get_value("IMPALA_PASSWORD"),
+        kerberos_service_name=configuration.get_value("IMPALA_KERBEROS_SERVICE_NAME"),
+        krb_host=configuration.get_value("IMPALA_KRB_HOST"),
+        use_http_transport=configuration.get_value("IMPALA_USE_HTTP_TRANSPORT"),
+        http_path=configuration.get_value("IMPALA_HTTP_PATH"),
+        auth_cookie_names=configuration.get_value("IMPALA_AUTH_COOKIE_NAMES"),
+        retries=configuration.get_value("IMPALA_RETRIES"),
+    )
+
+
+@hookimpl
+def vdk_configure(config_builder: ConfigurationBuilder) -> None:
+    """
+    Here we define what configuration settings are needed for Impala with reasonable defaults
+    """
+    config_builder.add(
+        key="IMPALA_HOST",
+        default_value="localhost",
+        description="The Impala host we want to connect to.",
+    )
+    config_builder.add(
+        key="IMPALA_PORT",
+        default_value=21050,
+        description="The Impala port we want to connect to.",
+    )
+    config_builder.add(key="IMPALA_DATABASE", default_value=None, description="")
```

Comment:
For descriptions, you can find details in impala.dbapi connect() pydocs

Reply:
Thank you for the suggestion.